neqo-transport/src/connection/mod.rs

```diff
@@ -2000,7 +2000,10 @@ impl Connection {
         // Count how many bytes in this range are non-zero.
         let pn_len = mem::size_of::<PacketNumber>()
             - usize::try_from(unacked_range.leading_zeros() / 8).unwrap();
-        // pn_len can't be zero (unacked_range is > 0)
+        assert!(
+            pn_len > 0,
+            "pn_len can't be zero as unacked_range should be > 0, pn {pn}, largest_acknowledged {largest_acknowledged:?}, tx {tx}"
+        );
```

Comment:
Instead of panicing lower down in the call stack in `Encoder::encode_uint`, panic here already, given that more metadata is available.

https://github.com/mozilla/neqo/blob/d513712bc9d06e297af5cf317a9f6dd70e69a88b/neqo-common/src/codec.rs#L309